Fix: Invite Group create modal forms accessibility

deepsidhu85 opened the pull request:
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in visual indicators for required fields and error display with informative error messages when fields are left blank.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

<img width="2560" height="1473" alt="image" src="https://github.com/user-attachments/assets/ab8d6719-a703-499b-84c7-2f0c1dc28923" />

<img width="2560" height="1209" alt="image" src="https://github.com/user-attachments/assets/ce0528d2-2698-48f1-8ff3-1cc09dda9f73" />

<img width="2560" height="1473" alt="image" src="https://github.com/user-attachments/assets/680edf4a-fb75-4292-8aee-dc52daccc32e" />


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Try to share a namespace with a group (Members -> Invite Group) to a project and group but leave the access level blank
2. Verify that error messages are displayed and linked to the inputs via aria-describedby
3. The `Share` button is disabled if the group isn't selected. From the chrome inspect, remove the disabled attribute on the button and click the button
4. Verify that error messages are displayed and linked to the inputs via aria-describedby
5. Repeat 1-4 but this time share namespace with a group successfully, then try to share the namespace with the same group

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [X] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

joshsadam reviewed (commented):
Ran into an issue where I was left in a state that I could not submit the form and it did not give me any feedback.

<img width="1256" height="727" alt="Image" src="https://github.com/user-attachments/assets/3537a293-b3f5-43af-b5b8-54ca5ebdee0f" />

To replicate

1. I was using the sample data
2. I was in the group Yersinia
3. I set the group to Bacillus,
4. Set the access level
5. Submitted the form

deepsidhu85 commented:
> Ran into an issue where I was left in a state that I could not submit the form and it did not give me any feedback.
> 
> <img alt="Image" width="1256" height="727" src="https://private-user-images.githubusercontent.com/11295750/477799921-3537a293-b3f5-43af-b5b8-54ca5ebdee0f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxODE1NjMsIm5iZiI6MTc1NTE4MTI2MywicGF0aCI6Ii8xMTI5NTc1MC80Nzc3OTk5MjEtMzUzN2EyOTMtYjNmNS00M2FmLWI1YjgtNTRjYTVlYmRlZTBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE0VDE0MjEwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwMjA1Yjk3MDRjOTNiYmY0NWNkN2JjYzZmY2ZkNTEyN2VlZjM4NDNjMjI5OWRhY2NjZDJjZjM1OTY3YWI1MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HpJSm8kJrWR6o2jkCTtw7AtaeuaJLLPFzPqZBty_lC8">
> To replicate
> 
> 1. I was using the sample data
> 2. I was in the group Yersinia
> 3. I set the group to Bacillus,
> 4. Set the access level
> 5. Submitted the form

Fixed in e8f8da08ed62c61e024f4f9c1ba0b8b15109ff68. The errors were being displayed for the `group` error attribute when it should've been `group_id`

deepsidhu85 commented:
> > > Ran into an issue where I was left in a state that I could not submit the form and it did not give me any feedback.
> > > <img alt="Image" width="1256" height="727" src="https://private-user-images.githubusercontent.com/11295750/477799921-3537a293-b3f5-43af-b5b8-54ca5ebdee0f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxODE1NjMsIm5iZiI6MTc1NTE4MTI2MywicGF0aCI6Ii8xMTI5NTc1MC80Nzc3OTk5MjEtMzUzN2EyOTMtYjNmNS00M2FmLWI1YjgtNTRjYTVlYmRlZTBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE0VDE0MjEwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwMjA1Yjk3MDRjOTNiYmY0NWNkN2JjYzZmY2ZkNTEyN2VlZjM4NDNjMjI5OWRhY2NjZDJjZjM1OTY3YWI1MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HpJSm8kJrWR6o2jkCTtw7AtaeuaJLLPFzPqZBty_lC8">
> > > To replicate
> > > 
> > > 1. I was using the sample data
> > > 2. I was in the group Yersinia
> > > 3. I set the group to Bacillus,
> > > 4. Set the access level
> > > 5. Submitted the form
> > 
> > 
> > Fixed in [e8f8da0](https://github.com/phac-nml/irida-next/commit/e8f8da08ed62c61e024f4f9c1ba0b8b15109ff68). The errors were being displayed for the `group` error attribute when it should've been `group_id`
> 
> I think your fix broke the group is empty error (worked on previous commit) <img alt="image" width="1828" height="914" src="https://private-user-images.githubusercontent.com/82407232/478060292-f845fbec-17db-4e8a-96c4-b76e1ea937a6.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTUxODM0NjksIm5iZiI6MTc1NTE4MzE2OSwicGF0aCI6Ii84MjQwNzIzMi80NzgwNjAyOTItZjg0NWZiZWMtMTdkYi00ZThhLTk2YzQtYjc2ZTFlYTkzN2E2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODE0VDE0NTI0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3OGIzNTU0MzY0NTg0YTI3ZmE1MjM3MjdlNDlhNDVkNTEwZDVmZWRlNmE2ODVkNzAzZDczNjdmNGNhNjcwYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.LoD-BzgROQeysFq0Oe8nSBJO-muOOjILHEN4qxOMrag">
> 
> Also just wondering if the error msg `Group access level provided is not included in the list of valid access levels` is required? Since we're providing users the options and they can't manually type in an option.

It would've really helped if I updated the invite modal for project group links as well! Updated in 9b21582ffd7209bf59165af00a80f42d6ab4bdf2. As for that error message, it is an error message set by the model validation, so going to leave it as is

joshsadam reviewed (approved):
This looks good to me. We still need to work on fixing those select inputs
